pykylin/cursor.py

```diff
@@ -17,15 +24,51 @@ def __init__(self, connection):
         self.fetched_rows = 0
 
     def callproc(self):
-        raise('Stored procedures not supported in Kylin')
+        raise ('Stored procedures not supported in Kylin')
 
     def close(self):
         logger.debug('Cursor close called')
 
+    @staticmethod
+    def trans_sql_for_kylin(sql_str):
+        sql_str = sql_str.lower()
+        rep_sql_list = rep_sql_regx.findall(sql_str)
+        if not rep_sql_list:
+            return sql_str
+        transformed_sql = ''
+        rep_sql_dict = dict()
+        rep_sql_set = set()
+        for rep_sql in rep_sql_list:
+            rep_sql_dict[rep_sql] = rep_sql.split('|')[-2]
+            rep_sql_set.add(rep_sql.split('|')[1])
+
+        # step 1, replace all the const str to data name
```

Comment:
These hand-written transforming steps tends to be very tricky.
SQLAlchemy's visitor pattern could probably help you.